setup.py

```diff
@@ -295,6 +295,7 @@ def core_requires():
         "langchain>=0.0.286",
         "SQLAlchemy",
         "pymysql",
+        "psycopg2"
```

Comment:
The packages in `setup_spec.extras["framework"]` is base dependencies of DB-GPT project,  can you move `psycopg2` into `setup_spec.extras["datasource"]`, then we can install with command `pip install -e ".[datasource]"`.

Reply:
@fangyinc done